packages/labs/virtualizer/README.md

```diff
@@ -154,7 +154,129 @@ html`
 
 ### Using the `grid` layout
 
-TODO
+The grid layout lays out uniformly sized items in a regular grid. You specify an `itemSize`, which is either strictly observed or (if you use the flex option) matched as closely as possible while taking up all space along the layout's inline axis. [You can view an example of the grid layout here](https://lit.dev/playground/#gist=a5960ffd0cb67c61083b4df0f9372eeb).
+
+_Note: When using the grid layout, Virtualizer explicitly sets the dimensions of its child elements. This differs from the behavior of the default flow layout._
+
+```js
+  render() {
+    return html`
+      <lit-virtualizer
+        .layout=${grid({
+          itemSize: {width: '100px', height: '100px'}
+        })}
+        .items=${this.photos}
+        .renderItem=${photos => html`<img src=${photo.url}>`}
+      ></lit-virtualizer>
+    `;
+  }
+```
+
+#### Options
+
+The following options for available for the `grid` layout.
+
+#### gap
+
+The `gap` property determines spacing between items and can be either a single pixel value such as `8px` (the default), or by applying both a vertical and horizontal pair such as `8px 0`.
+
+Additionally `auto` can be used to split available space evenly between items, however this may only make sense for inline axis.
```

Comment:
Regarding "however this may only make sense for inline axis" does `auto` even do anything for the block axis? If not, I'd say "however this only works for the inline axis". If it _does_ do something for the block axis it'd be good to say what that is.

Reply:
Tried to check on this, but I am running into some issues. I based this particular example on the playground, but it seems like the `auto` property may be bugged, or I'm just misunderstanding how it works.

```js
  render() {
    return html`
      <lit-virtualizer
        .layout=${grid({
          itemSize: {width: '100px', height: '100px'},
          gap: '0 auto',
          justify: 'space-between'
        })}
        .items=${this.photos}
        .renderItem=${photos => html`<img src=${photo.url}>`}
      ></lit-virtualizer>
    `;
  }
```

The following produces this error: ` Error: grid layout: column-gap cannot be set to 'auto' when direction is set to vertical`, but from what I can tell `direction` doesn't actually work for a grid layout, as changing it at all causes the virtualizer to disappear.

Perhaps I should omit documenting the auto property for now?